src/core/memory.cc

```diff
@@ -1803,6 +1805,8 @@ void on_allocation_failure(size_t size) {
         seastar_logger.error("Failed to allocate {} bytes", size);
         abort();
     }
+
+    alloc_stats::increment(alloc_stats::types::failed_allocs);
```

Comment:
This is the only change on the memory allocation path, and it is _far_ outside of the hot path :).

Reply:
why not before the if above? so even if you're running with abort-on-alloc-failure option, you'll still see the failure if you're observing the metric over monitoring.

Reply:
@raphaelsc - good question, I actually recall considering this but don't recall my reasoning. The chance you'll see the 0 -> 1 bump in monitoring is microscopic in a typical config, since a metrics scrape/push would have to happen in the small fraction of a second between the increment and the `abort()`, but I have no objection to moving it up. Do you think it would be better?

Reply:
I think one argument in favor of moving it up even above the `maybe_dump_memory_diagnostics` is that we might plausibly want to dump this in the diagnostics too.

Reply:
@raphaelsc - I've moved the increment to the top of the function.

I've also added the number of failed allocations to the memory diagnostics output, since it seems important enough, and added a unit test which checks that allocation failures increase the value reported in the diagnostics.

Reply:
It's not on the top of the function, maybe you moved it on the wrong commit.

Reply:
@avikivity - I moved it from the bottom to the top in a new commit which introduced also the listing of hard failure count into the diagnostics, but I've changed that now to put it at the top in the original commit.